TST, MAINT: 32-bit Linux test failures in wheels repo

As assessed in this wheels repo PR where I'm starting to get ready for `1.9.0rc1` by probing the wheel build/test status: https://github.com/MacPython/scipy-wheels/pull/166 

Two failing tests are reported on 32-bit Linux:

```
FAILED optimize/tests/test_linprog.py::TestLinprogHiGHSMIP::test_mip1
FAILED optimize/tests/test_milp.py::test_milp_options
```

Both failures appear to be segfaults:

<details>

<summary>
`test_milp_options` segfault trace
</summary>

```
Thread 0xf77f1b40 (most recent call first):
  File "/venv/lib/python3.9/site-packages/execnet/gateway_base.py", line 400 in read
  File "/venv/lib/python3.9/site-packages/execnet/gateway_base.py", line 432 in from_io
  File "/venv/lib/python3.9/site-packages/execnet/gateway_base.py", line 967 in _thread_receiver
  File "/venv/lib/python3.9/site-packages/execnet/gateway_base.py", line 220 in run
  File "/venv/lib/python3.9/site-packages/execnet/gateway_base.py", line 285 in _perform_spawn

Current thread 0xf7d08980 (most recent call first):
  File "/venv/lib/python3.9/site-packages/scipy/optimize/_milp.py", line 355 in milp
  File "/venv/lib/python3.9/site-packages/scipy/optimize/tests/test_milp.py", line 69 in test_milp_options
  File "/venv/lib/python3.9/site-packages/_pytest/python.py", line 192 in pytest_pyfunc_call
  File "/venv/lib/python3.9/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/venv/lib/python3.9/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/venv/lib/python3.9/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/venv/lib/python3.9/site-packages/_pytest/python.py", line 1761 in runtest
  File "/venv/lib/python3.9/site-packages/_pytest/runner.py", line 166 in pytest_runtest_call
  File "/venv/lib/python3.9/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/venv/lib/python3.9/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/venv/lib/python3.9/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/venv/lib/python3.9/site-packages/_pytest/runner.py", line 259 in <lambda>
  File "/venv/lib/python3.9/site-packages/_pytest/runner.py", line 338 in from_call
  File "/venv/lib/python3.9/site-packages/_pytest/runner.py", line 258 in call_runtest_hook
  File "/venv/lib/python3.9/site-packages/_pytest/runner.py", line 219 in call_and_report
  File "/venv/lib/python3.9/site-packages/_pytest/runner.py", line 130 in runtestprotocol
  File "/venv/lib/python3.9/site-packages/_pytest/runner.py", line 111 in pytest_runtest_protocol
  File "/venv/lib/python3.9/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/venv/lib/python3.9/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/venv/lib/python3.9/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/venv/lib/python3.9/site-packages/xdist/remote.py", line 110 in run_one_test
  File "/venv/lib/python3.9/site-packages/xdist/remote.py", line 91 in pytest_runtestloop
  File "/venv/lib/python3.9/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/venv/lib/python3.9/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/venv/lib/python3.9/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/venv/lib/python3.9/site-packages/_pytest/main.py", line 322 in _main
  File "/venv/lib/python3.9/site-packages/_pytest/main.py", line 268 in wrap_session
  File "/venv/lib/python3.9/site-packages/_pytest/main.py", line 315 in pytest_cmdline_main
  File "/venv/lib/python3.9/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/venv/lib/python3.9/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/venv/lib/python3.9/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/venv/lib/python3.9/site-packages/xdist/remote.py", line 291 in <module>
  File "/venv/lib/python3.9/site-packages/execnet/gateway_base.py", line 1554 in serve
  File "<string>", line 8 in <module>
  File "<string>", line 1 in <module>
[gw2] node down: Not properly terminated

```

</details>

<details>

<summary>
`test_mip1` segfault trace
</summary>

```
Thread 0xf77a2b40 (most recent call first):
  File "/venv/lib/python3.9/site-packages/execnet/gateway_base.py", line 400 in read
  File "/venv/lib/python3.9/site-packages/execnet/gateway_base.py", line 432 in from_io
  File "/venv/lib/python3.9/site-packages/execnet/gateway_base.py", line 967 in _thread_receiver
  File "/venv/lib/python3.9/site-packages/execnet/gateway_base.py", line 220 in run
  File "/venv/lib/python3.9/site-packages/execnet/gateway_base.py", line 285 in _perform_spawn

Current thread 0xf7cb9980 (most recent call first):
  File "/venv/lib/python3.9/site-packages/scipy/optimize/_linprog_highs.py", line 363 in _linprog_highs
  File "/venv/lib/python3.9/site-packages/scipy/optimize/_linprog.py", line 598 in linprog
  File "/venv/lib/python3.9/site-packages/scipy/optimize/tests/test_linprog.py", line 2211 in test_mip1
  File "/venv/lib/python3.9/site-packages/_pytest/python.py", line 192 in pytest_pyfunc_call
  File "/venv/lib/python3.9/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/venv/lib/python3.9/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/venv/lib/python3.9/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/venv/lib/python3.9/site-packages/_pytest/python.py", line 1761 in runtest
  File "/venv/lib/python3.9/site-packages/_pytest/runner.py", line 166 in pytest_runtest_call
  File "/venv/lib/python3.9/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/venv/lib/python3.9/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/venv/lib/python3.9/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/venv/lib/python3.9/site-packages/_pytest/runner.py", line 259 in <lambda>
  File "/venv/lib/python3.9/site-packages/_pytest/runner.py", line 338 in from_call
  File "/venv/lib/python3.9/site-packages/_pytest/runner.py", line 258 in call_runtest_hook
  File "/venv/lib/python3.9/site-packages/_pytest/runner.py", line 219 in call_and_report
  File "/venv/lib/python3.9/site-packages/_pytest/runner.py", line 130 in runtestprotocol
  File "/venv/lib/python3.9/site-packages/_pytest/runner.py", line 111 in pytest_runtest_protocol
  File "/venv/lib/python3.9/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/venv/lib/python3.9/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/venv/lib/python3.9/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/venv/lib/python3.9/site-packages/xdist/remote.py", line 110 in run_one_test
  File "/venv/lib/python3.9/site-packages/xdist/remote.py", line 91 in pytest_runtestloop
  File "/venv/lib/python3.9/site-packages/execnet/gateway_base.py", line 1554 in serve
  File "<string>", line 8 in <module>
  File "<string>", line 1 in <module>
[gw0] node down: Not properly terminated

```

</details>

The segfault in `scipy/optimize/_milp.py` seems to be from a call to C++ code via `_highs_wrapper(..)`.

The segfault in `scipy/optimize/_linprog_highs.py` similarly seems to result from `_highs_wrapper(..)`.

The raw log is here (but may disappear after a week or so): https://dev.azure.com/numpy/27346c6a-2774-4eac-bf85-e068127c0ccc/_apis/build/builds/24283/logs/53

@mckib2 and @mdhaber maybe for thoughts?

Interestingly, `test_mip1` doesn't seem to have any xfail/skip decorations, but `test_milp_options` does have an ominous marker:

`@pytest.mark.xfail(strict=True, reason="Needs to be fixed in _highs_wrapper")`

So, perhaps for the latter case we could skip execution entirely on 32-bit until the C++ stuff is fixed up, given that there are apparently issues there already even on 64-bit?

> @pytest.mark.xfail(strict=True, reason="Needs to be fixed in _highs_wrapper")

@mdhaber  I have no recollection of this `xfail` marking, but it could be that `_highs_wrapper` is (as a compiled library) having trouble passing stdout/stderr to the `capsys`?  When I run locally, I get `captured   = CaptureResult(out='', err='')` but I do get a bunch of stuff printed in the `Captured stdout call` output:
```
------------------------------------------------------------------------------------------ Captured stdout call -------------------------------------------------------------------------------------------

Presolve is switched off

Solving MIP model with:
   26 rows
   81 cols (81 binary, 0 integer, 0 implied int., 0 continuous)
   378 nonzeros

( 0.0s) Starting symmetry detection
( 0.0s) No symmetry present

Solving root node LP relaxation

        Nodes      |    B&B Tree     |            Objective Bounds              |  Dynamic Constraints |       Work      
     Proc. InQueue |  Leaves   Expl. | BestBound       BestSol              Gap |   Cuts   InLp Confl. | LpIters     Time

         0       0         0   0.00%   1.730550595     inf                  inf        0      0      0        33     0.0s
 R       0       0         0   0.00%   1.945020539     4.571417633       57.45%       68      4      0        55     0.0s

Solving report
  Status            Time limit reached
  Primal bound      4.57141763276
  Dual bound        2.42641568297
  Solution status   feasible
                    4.57141763276 (objective)
                    0 (bound viol.)
                    0 (int. viol.)
                    0 (row viol.)
  Timing            0.05 (total)
                    0.00 (presolve)
                    0.00 (postsolve)
  Nodes             0
  LP iterations     335 (total)
                    0 (strong br.)
                    302 (separation)
                    0 (heuristics)
========================================================================================= short test summary info =========================================================================================
FAILED scipy/optimize/tests/test_milp.py::test_milp_options - AssertionError: assert 'Presolve is switched off' in ''
```

--- a/scipy/optimize/tests/test_linprog.py
+++ b/scipy/optimize/tests/test_linprog.py
@@ -2,6 +2,7 @@
 Unit test for Linear Programming
 """
 import sys
+import platform
 
 import numpy as np
 from numpy.testing import (assert_, assert_allclose, assert_equal,
@@ -2199,6 +2200,10 @@ class TestLinprogHiGHSMIP():
     method = "highs"
     options = {}
 
+    @pytest.mark.xfail(condition=(sys.maxsize < 2 ** 32 and
+                       platform.system() == "Linux"),
+                       run=False,
+                       reason="gh-16347")
     def test_mip1(self):
         # solve non-relaxed magic square problem (finally!)
         # also check that values are all integers - they don't always

--- a/scipy/optimize/tests/test_milp.py
+++ b/scipy/optimize/tests/test_milp.py
@@ -57,8 +57,10 @@ def test_milp_iv():
         milp([1, 2, 3], bounds=([1, 2, 3], [set(), 4, 5]))
 
 
-@pytest.mark.xfail(strict=True, reason="Needs to be fixed in `_highs_wrapper`")
+@pytest.mark.xfail(run=False,
+                   reason="Needs to be fixed in `_highs_wrapper`")
 def test_milp_options(capsys):
+    # run=False now because of gh-16347
     message = "Unrecognized options detected: {'ekki'}..."
     options = {'ekki': True}
     with pytest.warns(RuntimeWarning, match=message):

